plug-ins/equalabel.ny

```diff
@@ -23,134 +23,145 @@ $copyright (_ "GNU General Public License v2.0 or later")
           (format nil (_ "~a~a ~a at intervals of ~a seconds.~%")
-                  msg number labeltype interval))
-      (if (> region 0)
+                  msg number labeltype intervals))
+      (if (> REGION 0)
           (format t (_ "~aRegion length = ~a seconds.")
-                  msg region)
+                  msg REGION)
           (format t msg)))))
 
 
 (defun get-interval-count (&aux dur)
 "Number of labels when interval is specified"
   (setf dur (get-safe-duration))
-  (case adjust
+  (case ADJUST
     ;; Interval is user input value
-    (0  (let ((n (truncate (/ dur interval))))
-          (if (< (* n interval) dur)
+    (0  (let ((n (truncate (/ dur INTERVAL))))
+          (if (< (* n INTERVAL) dur)
               (1+ n)
               n)))
-    ;; Adjust interval to fit length
-    (1  (let* ((min-num (truncate (/ dur interval)))
+    ;; Adjust space between labels to fit length
+    (1  (let* ((min-num (truncate (/ dur INTERVAL)))
                (max-num (1+ min-num)))
           (if (and (> min-num 0)
-                   (< (abs (- interval (/ dur min-num)))
-                      (abs (- interval (/ dur max-num)))))
+                   (< (abs (- INTERVAL (/ dur min-num)))
+                      (abs (- INTERVAL (/ dur max-num)))))
               min-num
               max-num)))))
 
 
-(defun make-one-label (time num)
+(defun make-one-label (time num num-txt zeropad)
 "Make a single label"
   (let* ((num-text (format nil "~a" num))
          (non-zero-digits (length num-text)))
-    (if (= zeros 0)
+    (if (= zeropad 0)
         (setf num-text "")
-        (dotimes (i (max 0 (- zeros non-zero-digits)))
+        (dotimes (i (max 0 (- zeropad non-zero-digits)))
           (setf num-text (format nil "~a~a" "0" num-text))))
-    (if num-before-text
-      (setf text (format nil "~a~a" num-text labeltext))
-      (setf text (format nil "~a~a" labeltext num-text)))
-    (list time (+ time region) text)))
+    (if num-txt  ; Number before text in label.
+      (setf text (format nil "~a~a" num-text LABELTEXT))
+      (setf text (format nil "~a~a" LABELTEXT num-text)))
+    (list time (+ time REGION) text)))
+
 
 (defun lasttrackp ()
 "True when processing the final selected track"
   (let ((index (get '*track* 'index))
         (num (length (get '*selection* 'tracks))))
     (= index num)))
 
+
 (defun get-safe-duration ()
    "Returns a safe duration for the labels to be distributed in"
-   (let ((duration (- (get-duration 1) region)))
+   (let ((duration (- (get-duration 1) REGION)))
       (if (< duration 0)
           0
           duration)))
 
-(setf num-before-text (<= zeros 3))
-(setf zeros (1+ (rem (1- zeros) 3)))
+
+(setf num-before-text (<= ZEROS 3))
+(setf zeropad (1+ (rem (1- ZEROS) 3)))
 
 ;; Analyze plug-ins may return text message per track but
 ;; we only want error messages once, and only one set of labels.
 (if (lasttrackp)
-    (make-labels)
-    "") ;No-op
+    (make-labels num-before-text zeropad)
```

Comment:
Why does this one gain arguments?

Reply:
> Why does this one gain arguments?

The function `make-labels` was previously defined as: `(defun make-labels (&aux labels)`
The `&aux labels`  parameter defines an auxiliary variable that is local within the function.

`num-before-text` was previously defined with global scope in the line:
```
(setf num-before-text (<= zeros 3))
```
It is generally better to avoid global variables unless absolutely necessary (as is the case with `;control` lines).
By passing `num-before-text` as an argument, we maintain encapsulation rather than littering the code unnecessarily with globals.

In the case of `zeropad`, it was even worse as not only were we previously using `zeros` with global scope, but we were also modifying it within the function `make-one-label`. In the new version, we explicitly pass the value of `zeropad` and maintain encapsulation, which not only improves readability, but is also a lot safer.

Reply:
thanks for the explanations!